### PR TITLE
move iptables from pkg/util to component-helpers, and remove unused pkg/util/ipconfig

### LINKED ISSUE #92369: Move pkg/proxy to k8s.io/kube-proxy
**What would you like to be added**:

Could you please move pkg/proxy to k8s.io/kube-proxy as an individual module?

**Why is this needed**:

Some CNIs would like to implement their own kube-proxy like component to gain more performance and thus would like to reuse kube-proxy code.  e.g. https://github.com/vmware-tanzu/antrea/pull/772 .

### PR DESCRIPTION
**Pull request by aartij17**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Removes the pkg/util/ipconfig module as discussed in issue #92369.
This PR also moves the `pkg/util/iptables` module to `k8s.io/component-base/network`
#### Which issue(s) this PR fixes:

Fixes #92369 

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by spiffxp** [comment]
/test pull-kubernetes-node-e2e-containerd
ref: #105381

File: pkg/kubelet/kubelet_network.go [hunk 1]
```diff
@@ -19,10 +19,10 @@ package kubelet
 import (
 	"fmt"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (rikatz) [inline pkg/kubelet/kubelet_network.go:22]: vscode linter usually adds this v1, would suggest to remove it and keep as the old entry

File: pkg/util/iptablesbkp/testing/fake.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ import (
 	"strings"
```
Comment (rikatz) [inline pkg/util/iptablesbkp/testing/fake.go:22]: Let's remove this "iptablesbkp" folder

File: staging/src/k8s.io/component-helpers/network/utils/iptables/OWNERS [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# See the OWNERS docs at https://go.k8s.io/owners
```
Comment (rikatz) [inline staging/src/k8s.io/component-helpers/network/utils/iptables/OWNERS:1]: This file can be a copy of https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/OWNERS instead (actually it's better to copy it, and we end by using the proper aliases)

**Comment by rikatz** [comment]
/retitle WIP: move iptables from pkg/util to component-helpers, and remove unused pkg/util/ipconfig

**Review by rikatz — CHANGES_REQUESTED** [review]
Some minor changes, otherwise lgtm:

* OWNERS file can be consolidated in staging/src/k8s.io/component-helpers/network/OWNERS 
* Please remove the WIP and Draft status as soon as this is ready to review, so the CI can run.

Thanks!

**Comment by dims** [comment]
please reopen if you still need this WIP PR. Trying to declutter the open PR(s)

**Pull request closed without merging**